mlflow/genai/__init__.py

```diff
@@ -1,4 +1,46 @@
 from mlflow.genai.evaluation import evaluate, to_predict_fn
 from mlflow.genai.scorers import Scorer, scorer
 
-__all__ = ["evaluate", "to_predict_fn", "Scorer", "scorer"]
+try:
+    from mlflow.genai.labeling import (
+        Agent,
+        LabelingSession,
+        ReviewApp,
+        get_review_app,
+        create_labeling_session,
+        get_labeling_sessions,
+        get_labeling_session,
+        delete_labeling_session,
+    )
+    # Stick this at the end to avoid unnecessary warnings
+    from mlflow.genai.datasets import (
+        EvaluationDataset,
+        create_dataset,
+        delete_dataset,
+        get_dataset,
+    )
+except ImportError:
+    # Silently fail if the databricks-agents package is not installed
+    pass
```

Comment:
@smoorjani I tried out the branch, and `mlflow.genai.create_dataset` isn't available in the `mlflow.genai` namespace because the "library not installed" errors from importing things from within `mlflow.genai.labeling` are masked.

We should move `mlflow.genai.datasets` *outside* of the try / except ImportError block

Reply:
Can we also add test coverage?

Reply:
Moved the three methods out of the try/except + added test coverage.